ui/test/nuxeo-aggregation-widget.test.js

```diff
@@ -55,42 +131,235 @@ const data = {
+      const nuxeoCheckboxAggregation = await fixture(
+        html`
+          <nuxeo-checkbox-aggregation></nuxeo-checkbox-aggregation>
+        `,
+      );
+
+      const message = nuxeoCheckboxAggregation.shadowRoot.querySelector('span');
+      expect(isElementVisible(message)).to.be.true;
+      expect(message.innerHTML).to.equal('checkboxAggregation.noResults');
+    });
   });
 
-  suite('Given a nuxeo-checkbox-aggregation widget on document aggregations', () => {
-    let documentAggs;
+  suite('Non-collapsible', () => {
+    test('Should display a label when "label" is set', async () => {
+      const nuxeoCheckboxAggregation = await fixture(
+        html`
+          <nuxeo-checkbox-aggregation .data=${data} label="Some label"> </nuxeo-checkbox-aggregation>
+        `,
+      );
 
-    setup(async () => {
-      documentAggs = await fixture(html`
-        <nuxeo-checkbox-aggregation .data=${data}></nuxeo-checkbox-aggregation>
-      `);
+      const label = nuxeoCheckboxAggregation.shadowRoot.querySelector('label');
+      expect(label.innerHTML).to.equal('Some label');
     });
 
-    test('It displays the title of the document', () => {
-      const choices = dom(documentAggs.root).querySelectorAll('paper-checkbox');
-      expect(choices.length).to.be.equal(2);
-      expect(choices[0].textContent.trim()).to.be.equal('Tolkien (2)');
-      expect(choices[1].textContent.trim()).to.be.equal('Asimov (1)');
+    test('Should display the title of the document when "data" is set', async () => {
+      const nuxeoCheckboxAggregation = await fixture(
+        html`
+          <nuxeo-checkbox-aggregation .data=${data}></nuxeo-checkbox-aggregation>
+        `,
+      );
+
+      const checkboxes = nuxeoCheckboxAggregation.shadowRoot.querySelectorAll('paper-checkbox');
+      expect(checkboxes.length).to.equal(9);
+      const results = [
+        'Tolkien (2)',
+        'Asimov (1)',
+        'Hemingway (3)',
+        'Dostoevsky (4)',
+        'Tolstoy (5)',
+        'Pessoa (6)',
+        'Balzac (7)',
+        'Cervantes (8)',
+        'Shakespeare (9)',
+      ];
+      checkboxes.forEach((element, i) => {
+        expect(isElementVisible(element)).to.be.true;
+        expect(element.innerHTML.trim()).to.equal(results[i]);
+      });
+    });
+
+    test('Should sort checkboxes alphabetically when "sort-by-label" is set', async () => {
+      const nuxeoCheckboxAggregation = await fixture(
+        html`
+          <nuxeo-checkbox-aggregation .data=${data} sort-by-label></nuxeo-checkbox-aggregation>
+        `,
+      );
+
+      const checkboxes = nuxeoCheckboxAggregation.shadowRoot.querySelectorAll('paper-checkbox');
+      expect(checkboxes.length).to.equal(9);
+      expect(checkboxes[0].textContent.trim()).to.equal('Asimov (1)');
+      expect(checkboxes[1].textContent.trim()).to.equal('Balzac (7)');
+      expect(checkboxes[2].textContent.trim()).to.equal('Cervantes (8)');
```

Comment:
nit: I think we could prevent some duplication with:
```suggestion
      expect(Array.from(checkboxes).map((c) => c.innerText)).to.deep.equal(['Asimov (1)', 'Balzac (7)', 'Cervantes (8)']);
```